src/mod_muc_room.erl

```diff
@@ -3295,13 +3295,12 @@ change_config(Config, StateData) ->
     case {(StateData#state.config)#config.persistent,
 	  Config#config.persistent}
 	of
-      {_, true} ->
-	  mod_muc:store_room(NSD#state.server_host,
-			     NSD#state.host, NSD#state.room, make_opts(NSD));
-      {true, false} ->
-	  mod_muc:forget_room(NSD#state.server_host,
-			      NSD#state.host, NSD#state.room);
-      {false, false} -> ok
+      {_, _} ->
+	      mod_muc:forget_room(NSD#state.server_host,
+		      NSD#state.host, NSD#state.room),
+	      mod_muc:store_room(NSD#state.server_host,
+		      NSD#state.host, NSD#state.room, make_opts(NSD));
+	    {false, false} -> ok
```

Comment:
alignment seems off here.... and I'm not sure if this fallback is ever going to happen?

Reply:
Fixed the indentation.  And its possible, if it was never a persistent room in the past or if we're not setting it then it won't affect the temporary room.